### PR TITLE
[BPK-878] Add rn banner alert

### PR DESCRIPTION
**Pull request by georgegillams**

### CODE REVIEW

**Comment by jamesf3rguson** [comment]
Hey @georgegillams, I've spend some time reviewing this from a design pov. Here are my comments:

- [ ] Remove set height (to allow for multi-line)
- [ ] Use paddingTop: spacingSm
- [ ] Use paddingBottom: spacingSm
- [ ] Use paddingLeft: spacingMd
- [ ] Use paddingRight: spacingMd
- [ ] Add minHeight: spacingXl - 2
- [ ] Change text size to Sm
- [ ] Change value of text in storybook examples to something more meaningful. Probably worth following what we have for web if thats ok.
- [ ] Icon is too big. Can we use small icon size?
- [ ] Chevron icon should be Gray-700
- [ ] Close icon should be Gray-500
- [ ] Expanded text size should be Sm
- [ ] Expanded text colour should be Gray-500
- [ ] Tapping anywhere on a dismissible banner alert makes it disappear. Should only be the close icon

**Comment by jamesf3rguson** [comment]
Matt and I just noticed you need to rebase on master as you don't have any of the package lock files.

File: native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+  SUCCESS: 'success',
+  WARN: 'warn',
+  ERROR: 'error',
+};
+
+const styles = StyleSheet.create({
+  outerContainer: {
+      // TODO Replace with borderSm token once available
+    borderWidth: 1,
+      // TODO Replace with radiiSm token once available
```
Comment (shaundon) [inline native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js]: I think these have now been added, so probably good to replace them now!

File: native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+  const textComponent = (
+    <BpkText textStyle="sm" style={styles.text}>{message}</BpkText>
+  );
+
+  const actionComponent = (
+    <Image
+      style={buttonIconStyle}
+      source={buttonIconSource}
+    />
+  );
```
Comment (shaundon) [inline native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js]: Could this be made to use `react-native-bpk-component-icon`?

File: native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+} from 'react-native';
+import React from 'react';
+import PropTypes from 'prop-types';
+import BpkText from 'react-native-bpk-component-text';
+
+const tickIcon = require('./icons/lg/tick-circle.png'); // eslint-disable-line import/no-unresolved
+const informationIcon = require('./icons/lg/information-circle.png'); // eslint-disable-line import/no-unresolved
+const closeIcon = require('./icons/lg/close.png'); // eslint-disable-line import/no-unresolved
+const chevronUpIcon = require('./icons/lg/chevron-up.png'); // eslint-disable-line import/no-unresolved
+const chevronDownIcon = require('./icons/lg/chevron-down.png'); // eslint-disable-line import/no-unresolved
```
Comment (matteo-hertel) [inline native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js]: How about we use `BpkIcon` here 😄 ?

File: native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+    ALERT_TYPES.WARN,
+    ALERT_TYPES.ERROR,
+  ]).isRequired,
+  children: PropTypes.node,
+  dismissable: dismissablePropType,
+  expanded: PropTypes.bool,
+  onPress: PropTypes.func,
+  actionButtonLabel: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.func,
```
Comment (matteo-hertel) [inline native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js]: Why is a function allowed here? I can't find it in the docs

File: native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+  ]),
+  style: stylePropType,
+};
+
+BpkBannerAlert.defaultProps = {
+  children: null,
+  dismissable: false,
+  expanded: false,
+  onPress: () => null,
```
Comment (matteo-hertel) [inline native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js]: It looks like this function is applied to all the `TouchableHighlight` so while closing, while expanding etc, is this intended behaviour?

File: native/packages/react-native-bpk-component-banner-alert/stories.js [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+  android: () => require('bpk-tokens/tokens/android/base.react.native.common.js'), // eslint-disable-line global-require
+})();
+
+const {
+  colorGray500,
+  spacingSm,
+  spacingBase,
+} = tokens;
+
+const styles = StyleSheet.create({
```
Comment (matteo-hertel) [inline native/packages/react-native-bpk-component-banner-alert/stories.js]: There is a new standardised layout for the storybooks, can you double check with @shaundon on how to use it?

**Comment by georgegillams** [comment]
@shaundon how about that:
screen_shot_2017-10-17_at_09_34_01

File: native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js (hunk 2 above)
Comment (georgegillams) [reply native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js]: Not sure - I think we should discuss this at standup, and then create a PR to replace all PNGs with the icon component

**Comment by shaundon** [comment]
Screenshot looks perfect now!

File: native/packages/react-native-bpk-component-banner-alert/readme.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+npm install react-native-bpk-component-banner-alert --save-dev
+```
+
+## Usage
+
+```js
+import React, { Component } from 'react';
+import { View, StyleSheet } from 'react-native';
+import BpkText, {ALERT_TYPES} from 'react-native-bpk-component-text';
+import BpkBannerAlert, {ALERT_TYPES} from 'react-native-bpk-component-banner-alert';
```
[the remark this answers is not in the document]
Comment (georgegillams) [reply native/packages/react-native-bpk-component-banner-alert/readme.md]: Just a copy pasta in the line above :)

File: native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js (hunk 3 above)
Comment (georgegillams) [reply native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js]: ⬆️ https://github.com/Skyscanner/backpack/pull/156#discussion_r145050555

File: native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js (hunk 4 above)
Comment (georgegillams) [reply native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js]: @matteo-hertel I guess it should be string only?

File: native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js (hunk 5 above)
Comment (georgegillams) [reply native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js]: Expand / collapse etc should be left up to userland. The `onPress` function is there to allow them to respond to user interaction with the component

File: native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js (hunk 4 above)
Comment (matteo-hertel) [reply native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js]: As far as I remember it's string only

File: native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js (hunk 5 above)
Comment (matteo-hertel) [reply native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js]: my point is, there is one `onPress` and gets executed on both banner press and dismiss etc, I'm not sure you want the same function to run on both close and expand/collapse

File: native/packages/react-native-bpk-component-banner-alert/stories.js (hunk 6 above)
Comment (georgegillams) [reply native/packages/react-native-bpk-component-banner-alert/stories.js]: Sure - have done :)

**Comment by shaundon** [comment]
LGTM!

**Pull request merged**